### DISCUSSION TITLE
Bug? Nested field and onSubmit method

### DISCUSSION THREAD

**Discussion by rlesniak** [react-hook-form/react-hook-form#3849] [Q&A]
I have another troubles while moving away from Formik. Look at this [CDS](https://codesandbox.io/s/react-hook-form-validationschema-v6-ou04i?file=/src/index.js). I have Yup and simple nested field which can be unregistered as user wants.
My question is, why value of `additionalWebsite1` in `handleSubmit` is object, but in the next line `getValues()` returns real value which is `undefined`? Did I find bug?

**Comment by bluebill1049** [comment]
There is no such input named as `additionalWebsite1`, v6 doesn't look object key lookup. working v7 now and this feature will be unlocked.

**Reply by rlesniak** [reply]
But why results here are different
```
  const onSubmit = (data) => {
    console.log(data.additionalWebsite1 !== getValues("additionalWebsite1")); // WHY?
  };
```

**Reply by rlesniak** [reply]
Whats more interesting, when you comment line with `resolver`, value in onSubmit (`data.additionalWebsite1`) is perfectly fine.

**Reply by bluebill1049** [reply]
what's the steps to reproduce?

**Reply by rlesniak** [reply]
Just comment 22 line in CDS above, and submit.

**Reply by bluebill1049** [reply]
This is because we validate against what's in the schema that you defined: https://github.com/react-hook-form/resolvers/blob/master/src/yup.ts

**Reply by rlesniak** [reply]
Shouldn't validation influence on separate thing which are values? 
Second, how to make it work? Yup provide `when` so we can make dynamic fields.

**Reply by bluebill1049** [reply]
React hook form need to register those inputs, however in your case they never get registered. I would recommend just toggle the visibility, or use `shouldUnregister: false`. I would recommend the first option.
